src/graph.cpp

```diff
@@ -124,74 +113,7 @@ void Graph::batch_update(uint64_t src, const std::vector<uint64_t>& edges, Super
 }
 
 vector<set<node_t>> Graph::connected_components() {
-#ifdef USE_FBT_F
-  bf->force_flush(); // flush everything in buffertree to make final updates
-#else
-  wq->force_flush();
-#endif
-  GraphWorker::pause_workers(); // wait for the workers to finish applying the updates
-  // after this point all updates have been processed from the buffer tree
-  end_time = std::chrono::steady_clock::now();
-
-  printf("Total number of updates to sketches before CC %lu\n", num_updates.load()); // REMOVE this later
-  update_locked = true; // disallow updating the graph after we run the alg
-  bool modified;
-#ifdef VERIFY_SAMPLES_F
-  GraphVerifier verifier {cumul_in};
-#endif
-
-  do {
-    modified = false;
-    vector<node_t> removed;
-    for (node_t i: (*representatives)) {
-      if (parent[i] != i) continue;
-      boost::optional<Edge> edge = supernodes[i]->sample();
-#ifdef VERIFY_SAMPLES_F
-      if (edge.is_initialized())
-        verifier.verify_edge(edge.value());
-      else
-        verifier.verify_cc(i);
```

Comment:
I didn’t implement this in parallel because I assumed it was always going to be secondary to the single-threaded implementation. I think it won’t be too difficult to modify the impl of edge verify to wait until all queries to update CCs. I think verification is necessary to check correctness.

Reply:
Done